From ea2a0e128d71802f61b8da2e44de3c6325c5f328 Mon Sep 17 00:00:00 2001
From: Andrew Deason <adeason@sinenomine.net>
Date: Sat, 4 Apr 2020 22:35:07 -0500
Subject: [PATCH 03/12] Use autoconf-archive m4 from src/external

Switch to using the m4 macros from autoconf-archive in our
src/external mechanism, instead of manually-copied versions in src/cf.
The src/external copy of ax_gcc_func_attribute.m4 is identical to the
existing copy in src/cf, so that should incur no changes. There are
also a few new macros pulled in, but they are currently unused.

Increase our AC_PREREQ in configure.ac to 2.64, to match the AC_PREREQ
in some of the new files.

Reviewed-on: https://gerrit.openafs.org/14135
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Cheyenne Wills <cwills@sinenomine.net>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit ca847ddf35e336a8bc3159ce4b26f0162417bbd5)

Change-Id: Ifa43b3869e426fada5bd925b0ae002a0f6436232
---
 LICENSE      | 15 +++++++++++++++
 configure.ac |  2 +-
 regen.sh     |  8 ++++++--
 3 files changed, 22 insertions(+), 3 deletions(-)

diff --git a/LICENSE b/LICENSE
index 92528897a..f200e2a02 100644
--- a/LICENSE
+++ b/LICENSE
@@ -415,3 +415,18 @@ src/cf/lib-pathname.m4 are covered by the following license:
   THIS SOFTWARE IS PROVIDED "AS IS" AND WITHOUT ANY EXPRESS OR IMPLIED
   WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF
   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+
+====================================================
+
+The files under src/external/autoconf-archive/m4/ are covered by the following
+license:
+
+  Copyright (c) 2008 Guido U. Draheim <guidod@gmx.de>
+  Copyright (c) 2011 Maarten Bosmans <mkbosmans@gmail.com>
+  Copyright (c) 2013 Gabriele Svelto <gabriele.svelto@gmail.com>
+  Copyright (c) 2014 Mike Frysinger <vapier@gentoo.org>
+
+  Copying and distribution of this file, with or without modification, are
+  permitted in any medium without royalty provided the copyright notice
+  and this notice are preserved.  This file is offered as-is, without any
+  warranty.
diff --git a/configure.ac b/configure.ac
index 8ca01268b..87ffd1a51 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,4 +1,4 @@
-AC_PREREQ([2.60])
+AC_PREREQ([2.64])
 AC_INIT([OpenAFS],[m4_esyscmd(build-tools/git-version .)],[openafs-bugs@openafs.org],[],[http://www.openafs.org/])
 AC_CONFIG_AUX_DIR([build-tools])
 AC_CONFIG_MACRO_DIR([src/cf])
diff --git a/regen.sh b/regen.sh
index 3ada36c6b..3ae1987f0 100755
--- a/regen.sh
+++ b/regen.sh
@@ -26,11 +26,15 @@ else
   exit 1
 fi
 
+M4_INCS="-I src/cf"
+M4_INCS="$M4_INCS -I src/external/rra-c-util/m4"
+M4_INCS="$M4_INCS -I src/external/autoconf-archive/m4"
+
 echo "Running aclocal"
 if which aclocal > /dev/null 2>&1; then
-  aclocal -I src/cf -I src/external/rra-c-util/m4
+  aclocal $M4_INCS
 elif which aclocal-1.10 > /dev/null 2>&1; then
-  aclocal-1.10 -I src/cf -I src/external/rra-c-util/m4
+  aclocal-1.10 $M4_INCS
 else
   echo "No aclocal found on your system (looked for aclocal & aclocal-1.10)"
   exit 1
-- 
2.36.1

